src/frontend/src/lib/services/address.services.ts

```diff
@@ -36,6 +37,17 @@ import type { Principal } from '@dfinity/principal';
 import { assertNonNullish, isNullish } from '@dfinity/utils';
 import { get } from 'svelte/store';
 
+export const listenTestnetNetworksToAddBtcTestnetAddresses = async (): Promise<void> => {
+	testnetsStore.subscribe(async (data) => {
```

Comment:
* Good point, I changed the approach. It's simpler, I don't know why I didn't think of it before.
* Also good point, done in the new approach.
* This could be done but at the UI and store level (in another PR). The store should indicate that the address is loading. Does the store have the meaning of "loading"? I see the type is `T | null | undefined`. Does one of them mean loading?

Reply:
> This could be done but at the UI and store level (in another PR). The store should indicate that the address is loading. Does the store have the meaning of "loading"? I see the type is T | null | undefined. Does one of them mean loading?

Good point, `undefined`. Maybe not stricto sensu "loading", rather "not yet loaded or loading" but, yes!

Reply:
I can use `undefined` to show a spinner when I do the UI then.

Reply:
Yeah. Altenartively, a skeleton given that the toggle for testnet find place in a popover menu and in the settings page.